explore/src/main/scala/explore/targets/TargetSummaryTable.scala

```diff
@@ -174,34 +184,55 @@ object TargetSummaryTable extends TableHooks:
             acceptLabel = "Yes, delete",
             position = DialogPosition.Top,
             accept = props.targets
-              .mod(_.filter((id, _) => !selectedRowsIds.contains(id))) *> TargetSummaryActions
-              .deleteTargets(selectedRowsIds, props.programId)
-              .runAsyncAndForget,
+              .mod(_.filter((id, _) => !selectedRowsIds.contains(id))) *>
+              table.toggleAllRowsSelected(false) *>
+              TargetSummaryActions
+                .deleteTargets(selectedRowsIds, props.programId)
+                .runAsyncAndForget,
             acceptClass = PrimeStyles.ButtonSmall,
             rejectClass = PrimeStyles.ButtonSmall,
             icon = Icons.SkullCrossBones.color("red")
           )
 
+        def onTextChange(e: ReactEventFromInput): Callback =
+          val files = e.target.files.toList
+          // set value to null so we can reuse the import button
+          (Callback(e.target.value = null) *> filesToImport.set(files)).when_(files.nonEmpty)
+
         <.div(
           props.renderInTitle(
             React.Fragment(
               <.div(
                 ExploreStyles.TableSelectionToolbar,
+                <.label(PrimeStyles.ButtonSmall,
+                        ^.cls     := "pl-compact p-component p-button p-fileupload",
+                        ^.htmlFor := "target-import",
+                        Icons.FileImport
+                ),
+                <.input(^.tpe     := "file",
+                        ^.onChange ==> onTextChange,
+                        ^.id      := "target-import",
+                        ^.name    := "file",
+                        ^.accept  := ".csv"
+                ),
```

Comment:
Clever to hide the button. 

It's strange to me that clicking on the label would open the dialog, but useful.

Reply:
It is a bit odd but it seems the standard way to customize the button
Note that prime react has a `FileUpload` component but it assumes the file is uploaded to a backend where we are here processing it locally

Reply:
Interesting. Thanks.

I was just looking at FileUpload. It seems like you could use it if you want and set `customUpload` to true and provide a `uploadHandler`. Not sure what extra it would give you or how hard it would be to use.  I could do a facade for it if you think it would be useful.

Reply:
I don't think we'll need it very soon